tensorflow_lite_support/python/task/audio/audio_classifier.py

```diff
@@ -0,0 +1,140 @@
+      classification_options_pb2.ClassificationOptions] = None
+
+
+class AudioClassifier(object):
+  """Class that performs classification on audio."""
+
+  def __init__(self, options: AudioClassifierOptions,
+               classifier: _CppAudioClassifier) -> None:
+    """Initializes the `AudioClassifier` object."""
+    # Creates the object of C++ AudioClassifier class.
+    self._options = options
+    self._classifier = classifier
+
+  @classmethod
+  def create_from_options(cls,
+                          options: AudioClassifierOptions) -> "AudioClassifier":
+    """Creates the `AudioClassifier` object from audio classifier options.
+
+    Args:
+      options: Options for the audio classifier task.
+    Returns:
+      `AudioClassifier` object that's created from `options`.
+    Raises:
+      status.StatusNotOk if failed to create `AudioClassifier` object from
+        `AudioClassifierOptions` such as missing the model. Need to import the
+        module to catch this error: `from pybind11_abseil
+        import status`, see
+        https://github.com/pybind/pybind11_abseil#abslstatusor.
+    """
+    # Creates the object of C++ AudioClassifier class.
+    proto_options = _ProtoAudioClassifierOptions()
+    proto_options.base_options.CopyFrom(
+        task_utils.ConvertToProtoBaseOptions(options.base_options))
+
+    # Updates values from classification_options.
+    if options.classification_options:
+      if options.classification_options.display_names_locale:
+        proto_options.display_names_locale = \
+          options.classification_options.display_names_locale
+      if options.classification_options.max_results:
+        proto_options.max_results = options.classification_options.max_results
+      if options.classification_options.score_threshold:
+        proto_options.score_threshold = \
+          options.classification_options.score_threshold
+      if options.classification_options.class_name_allowlist:
+        proto_options.class_name_allowlist.extend(
+            options.classification_options.class_name_allowlist)
+      if options.classification_options.class_name_denylist:
+        proto_options.class_name_denylist.extend(
+            options.classification_options.class_name_denylist)
+
+    classifier = _CppAudioClassifier.create_from_options(proto_options)
+
+    return cls(options, classifier)
+
+  def create_input_tensor_audio(self) -> tensor_audio.TensorAudio:
+    """Creates a TensorAudio instance to store the audio input.
+    Returns:
+        A TensorAudio instance.
+    """
+    return tensor_audio.TensorAudio(
+      audio_format=self.required_audio_format,
+      sample_count=self.required_input_buffer_size)
+
+  def create_input_audio_record(
+      self
+  ) -> (tensor_audio.TensorAudio, sd.InputStream):
+    """Creates an AudioRecord instance to record audio.
+    Returns:
+        An AudioRecord instance.
+    """
+    return audio_record.AudioRecord(self.create_input_tensor_audio())
```

Comment:
@kinaryml Please don't create the AudioRecord class but use the sounddevice.InputStream directly like in the [sample](https://github.com/khanhlvg/tflite_raspberry_pi/blob/b1c1e3e38d80ebc8f59e8fc47d24493954c90161/audio_classification/audio_classifier.py#L184). I'll update the samples on tensorflow/examples to follow the new API pattern next week.

Reply:
@khanhlvg Alright, I'll revert the changes.